clients/include/testing_reduction_strided_batched.hpp

```diff
@@ -0,0 +1,212 @@
+ * ************************************************************************ */
+
+#include "cblas_interface.hpp"
+#include "norm.hpp"
+#include "rocblas.hpp"
+#include "rocblas_init.hpp"
+#include "rocblas_math.hpp"
+#include "rocblas_random.hpp"
+#include "rocblas_test.hpp"
+#include "rocblas_vector.hpp"
+#include "unit.hpp"
+#include "utility.hpp"
+
+template <typename T, typename R>
+using rocblas_reduction_strided_batched_t = rocblas_status (*)(rocblas_handle handle,
+                                                               rocblas_int    n,
+                                                               const T*       x,
+                                                               rocblas_int    incx,
+                                                               rocblas_stride stridex,
+                                                               rocblas_int    batch_count,
+                                                               R*             result);
+
+template <typename T, typename R>
+void template_testing_reduction_strided_batched_bad_arg(
+    const Arguments& arg, rocblas_reduction_strided_batched_t<T, R> func)
+{
+    rocblas_int N = 100, incx = 1, batch_count = 5;
+
+    static const size_t safe_size = 100;
+
+    rocblas_local_handle handle;
+
+    //
+    // allocate memory on device
+    //
+    device_vector<T> dx(batch_count);
+    CHECK_HIP_ERROR(dx.memcheck());
+
+    R h_rocblas_result;
+
+    EXPECT_ROCBLAS_STATUS(
+        (func(handle, N, nullptr, incx, incx * N, batch_count, &h_rocblas_result)),
+        rocblas_status_invalid_pointer);
+
+    EXPECT_ROCBLAS_STATUS((func(handle, N, dx, incx, incx * N, batch_count, nullptr)),
+                          rocblas_status_invalid_pointer);
+
+    EXPECT_ROCBLAS_STATUS((func(nullptr, N, dx, incx, incx * N, batch_count, &h_rocblas_result)),
+                          rocblas_status_invalid_handle);
+}
+
+template <typename T, typename R>
+void template_testing_reduction_strided_batched(
+    const Arguments&                          arg,
+    rocblas_reduction_strided_batched_t<T, R> func,
+    void (*CBLAS_FUNC)(rocblas_int, const T*, rocblas_int, R*))
+{
+    rocblas_int    N = arg.N, incx = arg.incx, batch_count = arg.batch_count;
+    rocblas_stride stridex = arg.stride_x;
+
+    double               rocblas_error_1, rocblas_error_2;
+    rocblas_local_handle handle;
+    if(batch_count < 0)
+    {
+        device_strided_batch_vector<T> dx(10, 1, 10, 10);
+        CHECK_HIP_ERROR(dx.memcheck());
+        device_vector<R> dr(10);
+        CHECK_HIP_ERROR(dr.memcheck());
+        CHECK_ROCBLAS_ERROR(rocblas_set_pointer_mode(handle, rocblas_pointer_mode_device));
+        EXPECT_ROCBLAS_STATUS(func(handle, N, dx, incx, stridex, batch_count, dr),
+                              rocblas_status_invalid_size);
+        return;
+    }
+
+    // check to prevent undefined memory allocation error
+    if(N <= 0 || incx <= 0 || batch_count == 0)
+    {
```

Comment:
See above. The two cases should be merged together and then only the expected return status needs to be differentiated. This change is being made globally in rocBLAS as part of the early exit and argument checking refactor.

Reply:
Fixed